### PR TITLE
TZ from env, client log, alpine 3.14, path DB changed.

### PR DESCRIPTION
**Pull request by edgd1er**
May be too many change for a single PR. this is still a work in progress as I wish to run as user py-kms with specific uid/gid.
Please note that also database path has been changed to allow volume or file mounting: /db/pykms_database.db
+ 3 python libs updates due to security fix.

### CODE REVIEW

**Comment by simonmicro** [comment]
Okay, I'm still not 100% percent conform with the new `start.py` but whatever - let's ignore that. But still please integrate https://github.com/Py-KMS-Organization/py-kms/commit/4ee742d671ae853f600b90425c97b2080ff17b80 and https://github.com/Py-KMS-Organization/py-kms/commit/677916a3ec9964b119b6478f1a45e484bc603433 of the current upstream, because this makes the script at least some sort more readable (and you have somehow dropped the `os.makedirs()` instruction. After that: I'll hit "merge" faster than you can comment it! :smile:

File: docker/start.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  sqlite_process = None
+  # Build the command to execute
+  command = [PYTHON3, '-u', 'pykms_Server.py', os.environ.get('IP'), os.environ.get('PORT')]
+  for (arg, env) in argumentVariableMapping.items():
+    if env in os.environ and os.environ.get(env) != '':
+      command.append(arg)
+      command.append(os.environ.get(env))
+
+  if enableSQLITE:
+    loggersrv.info("Storing database file to %s" % dbPath)
```
Comment (simonmicro) [inline docker/start.py:55]: Please add https://github.com/Py-KMS-Organization/py-kms/commit/4ee742d671ae853f600b90425c97b2080ff17b80

File: docker/start.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+sqliteWebPath = '/home/sqlite_web/sqlite_web.py'
+enableSQLITE = os.path.isfile(sqliteWebPath) and os.environ.get('SQLITE', 'false').lower() == 'true'
+dbPath = os.path.join(os.sep, 'home', 'py-kms', 'db', 'pykms_database.db')
+log_level = os.getenv('LOGLEVEL', 'INFO')
+
+
+def start_kms_client():
+  if not os.path.isfile(dbPath):
+    # Start a dummy activation to ensure the database file is created
+    client_cmd = [PYTHON3, '-u', 'pykms_Client.py', os.environ.get('IP', "0.0.0.0"), os.environ.get('PORT', 1688),
```
Comment (simonmicro) [inline docker/start.py:32]: And also here https://github.com/Py-KMS-Organization/py-kms/commit/4ee742d671ae853f600b90425c97b2080ff17b80

**Review by simonmicro — CHANGES_REQUESTED** [review]

**Comment by edgd1er** [comment]
changes are implemented. as a side-note, I'm not sure being able to change listening port for kms and websqlite is very useful. Being in a container, ports are reassigned, except when network is inhost-mode.

**Comment by Matthew-Beckett** [comment]
> changes are implemented. as a side-note, I'm not sure being able to change listening port for kms and websqlite is very useful. Being in a container, ports are reassigned, except when network is inhost-mode.

Can you expand on that part about the port reassignment? Docker ports are configurable and static in my experience.

**Comment by edgd1er** [comment]
> > changes are implemented. as a side-note, I'm not sure being able to change listening port for kms and websqlite is very useful. Being in a container, ports are reassigned, except when network is in host-mode.
> 
> Can you expand on that part about the port reassignment? Docker ports are configurable and static in my experience.

for me, internal port = port exposed inside the container, external port is the port exposed on the host.

when using bridge mode, an "external" port is mapped to the "internal" port: docker run -p 1080:8080 ... => access through host_ip:1080
when using host mode, only the exposed port(s), formerly the internal one is used: docker run -d --network host .... => access through host_ip:8080

As far as I know, most usage are in bridge mode to have network isolation, as such, changing internal ports are useless.
kms_server has port assignement as it may also be executed without docker, and then ports assignements might be useful.

websqlite is defined only in the container, defining internal port (inside the container) has no use when running in bridge mode. 

The question: do we need to give the opportunity to define (and maintain) the "internals" ports ( kms, web sqlite) feature, knowing most of the time, the container will be running in bridge mode ?

I might be wrong assuming host mode is rarely used.

**Comment by Matthew-Beckett** [comment]
> > > changes are implemented. as a side-note, I'm not sure being able to change listening port for kms and websqlite is very useful. Being in a container, ports are reassigned, except when network is in host-mode.
> 
> > 
> 
> > Can you expand on that part about the port reassignment? Docker ports are configurable and static in my experience.
> 
> 
> 
> for me, internal port = port exposed inside the container, external port is the port exposed on the host.
> 
> 
> 
> when using bridge mode, an "external" port is mapped to the "internal" port: docker run -p 1080:8080 ... => access through host_ip:1080
> 
> when using host mode, only the exposed port(s), formerly the internal one is used: docker run -d --network host .... => access through host_ip:8080
> 
> 
> 
> As far as I know, most usage are in bridge mode to have network isolation, as such, changing internal ports are useless.
> 
> kms_server has port assignement as it may also be executed without docker, and then ports assignements might be useful.
> 
> 
> 
> websqlite is defined only in the container, defining internal port (inside the container) has no use when running in bridge mode. 
> 
> 
> 
> The question: do we need to give the opportunity to define (and maintain) the "internals" ports ( kms, web sqlite) feature, knowing most of the time, the container will be running in bridge mode ?
> 
> 
> 
> I might be wrong assuming host mode is rarely used.

I understand what you mean now, I agree with you.

**Comment by simonmicro** [comment]
It is just the same philosophy for the ports as also for your uid/gid reassignment. After all docker allows you to remap our default docker uid of 1000 to any other arbitrary id as you need. But beside that the port specification could be really interesting in case you have to use the host networking mode - e.g. in case you run on a swarm and do not want to utilize load-balancing and be only available on an specific node.

**Comment by simonmicro** [comment]
Okay, let's hope this does not break anything!

**Comment by simonmicro** [comment]
Aaand the Dockerfile is broken. @edgd1er please fix this and test it this time.

**Pull request merged**